__tests__/habits/habits.test.js

```diff
@@ -164,6 +169,51 @@ describe('API - Habit', () => {
 				}));
 	});
 
+	describe('Habit - Update', () => {
+		test('It should require authorization', () =>
+			request.put('/api/habit/52').then(response => {
+				expect(response.statusCode).toBe(401);
+			}));
+
+		test('If no data is passed a 400 status should be returned with error', done =>
+			request
+				.put(`/api/habit/${habitId}`)
+				.set('Authorization', token)
+				.send({ user: user.id })
+				.end((err, res) => {
+					if (err) throw err;
+					expect(res.status).toBe(400);
+					expect(res.body.name).toBe('Habit name field is required');
+					done();
+				}));
+
+		test('If an invalid habit ID is passed a 404 error should be returned', done =>
+			request
+				.put('/api/habit/something')
+				.set('Authorization', token)
+				.send(habit)
+				.end((err, res) => {
+					if (err) throw err;
+					expect(res.status).toBe(404);
+					expect(res.body.message).toBe('Habit not found');
+					done();
+				}));
+
+		test('If a valid habit ID is passed, that habit should be updated', done => {
```

Comment:
Can you test something like this:
```
test('If the same habit ID is passed, it should no longer exist', done => {
			habit.name = 'New habit updated';

			request
 				.put(`/api/habit/${habitId}`)
 				.set('Authorization', token)
 				.send(habit)
 				.end((err, res) => {
 					if (err) throw err;
 					expect(res.status).toBe(200);
 					expect(res.body.habit.name).toBe('New habit updated');
					expect(res.body.habit. description).toBe(habit.description);
 					expect(res.body.success).toBeTruthy();
 					done();
 				});
		});
```
You should test if the value is updated, but also test if old values are not changed. Sorry if I'm annoying with this comments

Reply:
Okey. no problem.

Reply:
any more changes? guys!!!